waku/v2/node/wakunode2.nim

```diff
@@ -92,10 +92,11 @@ proc removeContentFilters(filters: var Filters, contentFilters: seq[ContentFilte
     # Iterate filter entries to remove matching content topics
     for cf in f.contentFilters.mitems:
       # Iterate content filters in filter entry
-      cf.contentTopics.keepIf(proc (t: auto): bool = t notin unsubscribeTopics)
+      if cf.contentTopic in unsubscribeTopics:
+        cf.contentTopic = ""
     # make sure we delete the content filter
     # if no more topics are left
-    f.contentFilters.keepIf(proc (cf: auto): bool = cf.contentTopics.len > 0)
+    f.contentFilters.keepIf(proc (cf: auto): bool = cf.contentTopic.len > 0)
```

Comment:
Do we want to check the length of the contentTopic? Seems like different semantically from length of previous contentTopics

Reply:
Its actually checking that the contentTopic is not an empty String, cos its set to an empty string if it already exists in the sequence of  topics to unsubscribe.

Reply:
I've commented on similar logic in `waku_filter.nim`. In short, since we have simplified the nested structure, there is no need to do two passes anymore (where we first remove individual contentTopics and thereafter remove empty contentFilters). You can now directly operate on `f.contentFilters` and simply remove the full `contentFilter` if the corresponding `contentTopic` is in the `unsubscribeTopics` sequence.